### PR TITLE
Prevent sof module from being unloaded when a pcm is open

### PR DESCRIPTION
**Pull request by ranj063**
I said prevent SOF module in the title but it actually applies to the legacy driver as well.

### CODE REVIEW

**Comment by ranj063** [comment]
@plbossart commit message fixed as suggested.

File: sound/soc/soc-pcm.c [hunk 1]
```diff
@@ -513,6 +517,10 @@ static int soc_pcm_open(struct snd_pcm_substream *substream)
 		    !component->driver->ops->open)
 			continue;
 
+		if (component->driver->module_get_upon_open)
+			if (!try_module_get(component->dev->driver->owner))
```
Comment (plbossart) [inline sound/soc/soc-pcm.c]: maybe use the same pattern as before with an && instead of an additional if?

Comment (ranj063) [reply sound/soc/soc-pcm.c]: @plbossart fixed

**Comment by plbossart** [comment]
> Without knowing the full context, it looks a bit weird why the behaviour is not like this for all devices (why to refcount in probe). But better to have the discussion in upstream, code and description looks clean and this is much better solution than the previous ones that were specific to SOF.

see the ascii art I created at https://mailman.alsa-project.org/pipermail/alsa-devel/2019-February/145048.html, it explains the dependencies between PCI and machine driver.

**Comment by ranj063** [comment]
> the 3rd patch contains changes that should be in the second. we should't mix core and SOF changes.

@plbossart I just realized I posted the wrong patch for soc-pcm.c changes to alsa-devel. I will update it with a V2. Could you please ack it there?

**Comment by ranj063** [comment]
updated the PR now.

**Pull request merged**